Library/Homebrew/dev-cmd/pr-pull.rb

```diff
@@ -368,6 +371,30 @@ def download_artifact(url, dir, pr)
     end
   end
 
+  def pr_check_conflicts(name, repo, pr)
+    long_build_pr_files = GitHub.get_long_build_pull_requests_files(name, repo)
+    this_pr_files = GitHub.get_pr_files(name, repo, pr)
+
+    conflicts = []
+    this_pr_files.each do |this_pr_file|
+      long_build_pr_files.each do |long_build_pr_file|
+        if this_pr_file == long_build_pr_file["file"]
+          conflicts.append("#{tap.default_remote}/pull/#{long_build_pr_file["pr"]}")
+        end
+      end
+    end
+    return if conflicts.blank?
```

Comment:
I'd suggest inlining all this logic, removing the methods (except `get_pr_files`) and then we can work together to refactor this into minimal Ruby with e.g. `select` and `map`.

Reply:
Done